pages/k8s/charm-kubernetes-master.md

```diff
@@ -0,0 +1,629 @@
+in nagios. So for instance the hostname would be something like:
+
+```
+    juju-myservice-0
+```
+
+If you're running multiple environments with the same services in them
+this allows you to differentiate between them.
+
+[Back to table](#table-nagios_context)
+
+
+### proxy-extra-args
+
+
+<a id="proxy-extra-args-description"> </a>
+**Description:**
+
+Space separated list of flags and key=value pairs that will be passed as arguments to
+kube-proxy. For example a value like this:
+
+```
+  runtime-config=batch/v2alpha1=true profiling=true
+```
+
+will result in kube-apiserver being run with the following options:
+  --runtime-config=batch/v2alpha1=true --profiling=true
+
+[Back to table](#table-proxy-extra-args)
+
+
+### scheduler-extra-args
+
+
+<a id="scheduler-extra-args-description"> </a>
+**Description:**
+
+Space separated list of flags and key=value pairs that will be passed as arguments to
+kube-scheduler. For example a value like this:
+
+```
+  runtime-config=batch/v2alpha1=true profiling=true
+```
+
+will result in kube-scheduler being run with the following options:
+  --runtime-config=batch/v2alpha1=true --profiling=true
+
+[Back to table](#table-scheduler-extra-args)
+
+
+### snapd_refresh
+
+
+<a id="snapd_refresh-description"> </a>
+**Description:**
+
+How often snapd handles updates for installed snaps. Setting an empty
+string will check 4x per day. Set to "max" to delay the refresh as long
+as possible. You may also set a custom string as described in the
+'refresh.timer' section here:
+  https://forum.snapcraft.io/t/system-options/87
+
+[Back to table](#table-snapd_refresh)
+
+
+### sysctl
+
+
+<a id="sysctl-default"> </a>
+**Default:**
+
+```
+{ net.ipv4.conf.all.forwarding : 1, net.ipv4.neigh.default.gc_thresh1 : 128, net.ipv4.neigh.default.gc_thresh2 : 28672, net.ipv4.neigh.default.gc_thresh3 : 32768, net.ipv6.neigh.default.gc_thresh1 : 128, net.ipv6.neigh.default.gc_thresh2 : 28672, net.ipv6.neigh.default.gc_thresh3 : 32768, fs.inotify.max_user_instances : 8192, fs.inotify.max_user_watches: 1048576 }
+```
+
+
+[Back to table](#table-sysctl)
+
+
+<a id="sysctl-description"> </a>
+**Description:**
+
+YAML formatted associative array of sysctl values, e.g.:
+'{kernel.pid_max : 4194303 }'. Note that kube-proxy handles
+the conntrack settings. The proper way to alter them is to
+use the proxy-extra-args config to set them, e.g.:
+
+```
+  juju config kubernetes-master proxy-extra-args="conntrack-min=1000000 conntrack-max-per-core=250000"
+  juju config kubernetes-worker proxy-extra-args="conntrack-min=1000000 conntrack-max-per-core=250000"
+```
+
+The proxy-extra-args conntrack-min and conntrack-max-per-core can be set to 0 to ignore
+kube-proxy's settings and use the sysctl settings instead. Note the fundamental difference between
+the setting of conntrack-max-per-core vs nf_conntrack_max.
+
+[Back to table](#table-sysctl)
+
+
+
+<!-- CONFIG ENDS -->
+
+
+
+
+
+
+
+
+
+
+
+
+
```

Comment:
Drop 12 blank lines above, unless these come in when the auto-config section is generated.